pkg/agents/team2/config.go

```diff
@@ -0,0 +1,126 @@
+		utilities.
+    Observation:
+        Observation               Min                     Max
+        hp			              0.0                     100.0
+        floor	                  1                       inf
+        foodOnPlat                0    	                  100
+
+		Other observations should come from communication with other agents
+        savedAgents               0                       number of agent per florr
+		Note: the particular combination of the observations correspond to a particular state of
+		the agent
+	State:
+		We initially only define 3x3x3 (27) states for testing purposes
+
+		Num				hp			floor			foodOnPlat
+		0				61-100		1-30				61-100
+		1				61-100		1-30				31-60
+		2				61-100		1-30				0-30
+		3				61-100		31-60				61-100
+		...
+		26				0-30		>60					0-30
+    Action:
+		We separate the TakeFood action into several actions
+
+        Action					Comment
+        disregard food 			takeFood(0), sparing food for agents below me
+		satisfice with food 	takeFood(min), eat only food required for survival
+		satisfy with food   	takeFood(max), eat as manu as possible (get max hp increment)
+	Policy:
+		We associate a probability, or a policy, to a certain action in a certain state.
+    Reward:
+        For MVP, the reward is only related with to hp increment of the agent itself.
+	Q-table:
+		We associate a Q-value to each action under a particular state.
+    Starting State:
+        Initialise policies to a uniform distribution
+		Initialise Q-table to 0
+
+    Episode Termination:
+        Agent gets killed
+		or
+		Simulation terminated
+*/
+
+type actionSpace struct {
+	//TODO: actionId is redundant and may be removed in further version
+	actionId  []int
+	actionSet map[int]func(hp int) food.FoodType
+}
+type CustomAgent2 struct {
+	*infra.Base
+	stateSpace    [][][]int
+	actionSpace   actionSpace
+	policies      [][]float32
+	rTable        [][]float32
+	qTable        [][]float32
+	isPlatArrived bool
+}
+
+func InitTable(numStates int, numActions int) [][]float32 {
+	var table = make([][]float32, numStates)
+	for i := 0; i < numStates; i++ {
+		table[i] = make([]float32, numActions)
+	}
+	return table
+}
+
+func New(baseAgent *infra.Base) (infra.Agent, error) {
+	stateSpace := InitStateSpace(3, 3, 3)
+	actionSpace := InitActionSpace()
+	policies := InitPolicies(27, 3)
+	rTable := InitTable(27, 3)
+	qTable := InitTable(27, 3)
+	return &CustomAgent2{
+		Base:          baseAgent,
+		stateSpace:    stateSpace,
+		actionSpace:   actionSpace,
+		policies:      policies,
+		rTable:        rTable,
+		qTable:        qTable,
+		isPlatArrived: false,
+	}, nil
+}
+
+func (a *CustomAgent2) Run() {
+	//Communication & Observation
+	//communicate before platform arrives to my floor
+
+	/*
+		msg := *messages.NewBaseMessage(a.Floor())
+		a.SendMessage(1, msg)
+	*/
+
+	//Perform the following only when platform arrives
+	//NOTE: should let infra team add a func to see whether the plaftfrom has arrived or not
```

Comment:
Just as an FYI, you could possibly use `baseagent.CurrPlatFood()` as a proxy for this?

Reply:
I did it. the CheckState function uses CurrPlatFood to determine whether the platform has arrived or not.